agent/cache-types/connect_ca_leaf.go

```diff
@@ -108,7 +108,8 @@ func (c *ConnectCALeaf) Fetch(opts cache.FetchOptions, req cache.Request) (cache
 		return result, errors.New("invalid RootCA response type")
 	}
 	if roots.TrustDomain == "" {
-		return result, errors.New("cluster has no CA bootstrapped")
+		return result, errors.New("cluster has no CA bootstrapped: ensure Connect" +
+			" is enabled in server config")
```

Comment:
This is my attempt to make it more obvious to users what the cause is (likely) to be. We can't be more explicit since it's possible to hit this just due to timing before the CA bootstrap happens, especially in dev setups where CA bootstrap happens on every server restart and proxies might already be hitting the APIs before it does.